service/arm_templates/network_setup.json

```diff
@@ -0,0 +1,82 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "apiVersion": {
+      "type": "string",
+      "defaultValue": "2017-03-01",
+      "metadata": {
+        "description": "API version used by the Microsoft.Network/virtualNetworks resources."
+      }
+    },
+    "addressPrefix": {
+      "type": "string",
+      "defaultValue": "10.0.0.0/16",
+      "metadata": {
+        "description": "The main address block reserved for this virtual network."
+      }
+    },
+    "subnetMasterCIDR": {
+      "type": "string",
+      "defaultValue": "10.0.1.0/24",
+      "metadata": {
+        "description": "The CIDR block for the Kubernetes masters' subnet"
+      }
+    },
+    "subnetWorkerCIDR": {
+      "type": "string",
+      "defaultValue": "10.0.2.0/24",
+      "metadata": {
+        "description": "The CIDR block for the Kubernetes workers' subnet"
+      }
+    }
+  },
+  "variables": {
+    "virtualNetworkName": "[concat(uniquestring(resourceGroup().id), '-clustervn')]",
```

Comment:
Could there be multiple virtual network per resourceGroup? If so, we'll need to pass the clusterID to this template and use it in the virtualNetworkName instead.

Reply:
I think it's always good to include `clusterID`.

Reply:
What is `uniquestring`?

Reply:
> Could there be multiple virtual network per resourceGroup?

I'm afraid you'll have to check :)

Reply:
@kopiczko https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-template-functions-string#uniquestring

Reply:
@kopiczko On Azure level we can create multiple virtual networks per resourceGroup. What I'm note sure is, does the GiantSwarm design requires having multiple virtual networks per resourceGroup?

In any case, I think I'll just find a way to setup a unique virtualNetworkName. That way, that'll be future-proof ^^